ethergo/backends/anvil/anvil.go

```diff
@@ -121,7 +122,7 @@ func NewAnvilBackend(ctx context.Context, t *testing.T, args *OptionBuilder) *Ba
 	// to prevent old containers from piling up, we set a timeout to remove the container.
 	require.Nil(t, resource.Expire(args.expirySeconds))
 
-	address := fmt.Sprintf("%s:%s", "http://localhost", resource.GetPort("8545/tcp"))
+	address := fmt.Sprintf("%s:%s", "http://localhost", dockerutil.GetPort(resource, "8545/tcp"))
```

Comment:
Okay, I'm not sure how relevant this is, but I want to mention it because of all the work you guys did getting this ready in #1330 and #1334 is to uplaod the docker logs everywhere we use [TailContainerLogs](https://pkg.go.dev/github.com/synapsecns/sanguine/core/dockerutil#TailContainerLogs)

Especially in [anvil](https://pkg.go.dev/github.com/synapsecns/sanguine/ethergo/backends/anvil) + core[https://pkg.go.dev/github.com/synapsecns/sanguine/core/metrics/localmetrics].

We could then export these as workflow artifacts like we did in 1290 here:

https://github.com/synapsecns/sanguine/pull/1290/files#diff-38442ca3eaca86339f50860b9eaa0b38301559f536d66a37d7200a23c117821eR235-R246

<img width="1399" alt="image" src="https://github.com/synapsecns/sanguine/assets/83933037/3020636e-ce47-4393-b180-2e2e3a5e05cc">
 
and they'd be downoadable like [this](https://github.com/synapsecns/sanguine/actions/runs/6038167381):

<img width="774" alt="image" src="https://github.com/synapsecns/sanguine/assets/83933037/bb0e4559-b29c-493a-998d-416dbe94514c">

(As an aside, we could also do this w/ all the jaeger tracing stuff possibly, that's just a lot hell of a lot more work, considering we're moving in the direction of disabling jaeger in tests, this could really just be an otlp file)

This way, when flakes do occur, they'd be much easier to track down.

If this makes sense, would appreciate it if guys could make a new issue and just reference this otherwise curious to here why it wouldn't be helpful

Reply:
FYI, we could also directly embed this into https://pkg.go.dev/github.com/synapsecns/sanguine/core@v0.0.77/testsuite#TestSuite.TearDownSuite, but this would be a lot quicker to just do once at the end of the workflow. My reco:

1. Specify a `ARTIFACT_DIR`
2. Write all thse filets tehre
3. Zip + upload w/ ${matrix.package}/docker-logs.zip.

Note: while their exist actions like [this](https://github.com/marketplace/actions/collect-docker-logs) these won't work because we cleanup docker logs since we are [resource constrained](https://github.com/synapsecns/sanguine/pull/1280#issuecomment-1704401113)

Reply:
Update: implemented a thing here: https://github.com/synapsecns/sanguine/pull/1339/files#diff-678682767f2477de3e3c546746f8568b9a1942b2c647d32331d7e774b8ff8d9fR204-R227